rotkehlchen/chain/ethereum/modules/scroll_bridge/decoder.py

```diff
@@ -0,0 +1,369 @@
+            new_event_type=HistoryEventType.DEPOSIT,
+            amount=amount,
+            from_chain=ChainID.ETHEREUM,
+            to_chain=ChainID.SCROLL,
+        )
+
+        refund_event_idx, refund_amount = self._get_refund_event_index_and_amount(
+            decoded_events=context.decoded_events,
+            sender=sender,
+        )
+
+        if refund_event_idx is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        fee_amount = orig_amount - amount - refund_amount
+
+        spend_l2_fees_event = self._make_l2_fees_event(
+            context=context,
+            fee_amount=fee_amount,
+            sender=sender,
+        )
+        context.decoded_events[refund_event_idx] = spend_l2_fees_event
+
+        maybe_reshuffle_events(
+            ordered_events=[spend_l2_fees_event, bridge_event],
+            events_list=context.decoded_events,
+        )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_erc20_deposit_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes ERC20 deposits and withdrawals. (Bridging ERC20 tokens from and to ethereum)"""
+        tx_log = context.tx_log
+
+        if tx_log.topics[0] not in (DEPOSIT_ERC20, FINALIZE_WITHDRAW_ERC20):
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(tx_log.topics[3])
+        to_address = hex_or_bytes_to_address(tx_log.data[:32])
+
+        if not self.base.any_tracked([from_address, to_address]):
+            return DEFAULT_DECODING_OUTPUT
+
+        ethereum_token_address = hex_or_bytes_to_address(tx_log.topics[1])
+        asset = self.base.get_or_create_evm_token(ethereum_token_address)
+        raw_amount = hex_or_bytes_to_int(tx_log.data[32:64])
+        amount = asset_normalized_value(raw_amount, asset)
+
+        expected_event_type, new_event_type, from_chain, to_chain, _ = bridge_prepare_data(
+            tx_log=tx_log,
+            deposit_topics=(DEPOSIT_ERC20,),
+            source_chain=ChainID.ETHEREUM,
+            target_chain=ChainID.SCROLL,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        # Find the fee refund event and remove it
+        refund_event_idx, refund_amount = None, FVal(0)
+        for idx, event in enumerate(context.decoded_events):
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == from_address and
+                event.asset == self.eth and
+                event.address == L1_MESSENGER_PROXY
+            ):
+                refund_event_idx = idx
+                refund_amount = event.balance.amount
+                break
+        if refund_event_idx is not None:
+            del context.decoded_events[refund_event_idx]
+
+        # Find the corresponding transfer event and update it
+        for event in context.decoded_events:
```

Comment:
you are iterating twice the events. Can'y you di it in one loop and keeping the id of the refund event to remove it later?